it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,364 @@
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    prepareUpdateDatabaseMock();
+    testManager.createTable("");
+  }
+
+  @Test
+  public void testCreateTableShouldThrowExceptionWhenSpannerCreateInstanceFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getInstanceAdminClient().createInstance(any()).get())
+        .thenThrow(InterruptedException.class);
+    prepareCreateDatabaseMock();
+    prepareUpdateDatabaseMock();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldThrowExceptionWhenSpannerCreateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareCreateInstanceMock();
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    prepareUpdateDatabaseMock();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldThrowExceptionWhenSpannerUpdateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    when(spanner.getDatabaseAdminClient().updateDatabaseDdl(any(), any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldWorkWhenSpannerDoesntThrowAnyError()
+      throws ExecutionException, InterruptedException {
+    //   arrange
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    prepareUpdateDatabaseMock();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act
+    testManager.createTable(statement);
+
+    // assert
+    verify(spanner.getDatabaseAdminClient(), times(2))
```

Comment:
Are we really calling this twice? I thought that we only called `updateDatabaseDdl` once when creating the table. What's the second one for?

Reply:
The second is when we mock updateDatabaseDdl.get() in prepareUpdateDatabaseMock()

Reply:
Ok. Could you verify the other mocks as well? All of them are changing external state that will impact future calls, so it'll be good to know that the instance and database were created correctly. This seems to only verify that creating a table was done correctly.

I think doing it in this test is enough.